cmake/CMakeLists.txt

```diff
@@ -1092,7 +1092,7 @@ if (onnxruntime_USE_RKNPU)
 endif()
 if (onnxruntime_USE_NNAPI_BUILTIN)
     list(APPEND ORT_PROVIDER_FLAGS -DUSE_NNAPI=1)
-    list(APPEND ORT_PROVIDER_CMAKE_FLAGS -Donnxruntime_USE_NNAPI_BUILTIN=1)
+    list(APPEND ORT_PROVIDER_CMAKE_FLAGS -Donnxruntime_USE_NNAPI_BUILTIN=1 -Donnxruntime_ENABLE_RUNTIME_OPTIMIZATION_IN_MINIMAL_BUILD=1)
```

Comment:
Yes, I think it may be too late here. Need to add the compile definition in line 351 if that's enabled.
The intent is for ORT_ENABLE_RUNTIME_OPTIMIZATION_IN_MINIMAL_BUILD to be defined if and only if onnxruntime_ENABLE_RUNTIME_OPTIMIZATION_IN_MINIMAL_BUILD is set in an extended minimal build.
If we want to always enable runtime optimizations in an extended minimal build, that's fine too. I think that's a bigger change because ideally we would remove the onnxruntime_ENABLE_RUNTIME_OPTIMIZATION_IN_MINIMAL_BUILD option.
Another way is to add something like this:
```
if (onnxruntime_EXTENDED_MINIMAL_BUILD AND onnxruntime_USE_NNAPI_BUILTIN)
  set(onnxruntime_ENABLE_RUNTIME_OPTIMIZATION_IN_MINIMAL_BUILD ON)
endif()
```
earlier in this file (before line 351).

Reply:
The code should be using ORT_ENABLE_RUNTIME_OPTIMIZATION_IN_MINIMAL_BUILD in any #ifdef statements. `onnxruntime_ENABLE_RUNTIME_OPTIMIZATION_IN_MINIMAL_BUILD` is the cmake parameter name.

Reply:
What actually uses ORT_PROVIDER_CMAKE_FLAGS?

Reply:
That's a good question, couldn't find any other usages with grep.